kinvolk/lokomotive
# Add a certificate rotation command

# Add a certificate rotation command

This builds on top of https://github.com/kinvolk/lokomotive/pull/1201.
It checks off the todo items and moves it into a separate command as well as add support for certificate rotation of etcd on all platforms. 
Currently it will not rotate the private keys as that will cause some issues with the trust chain. It also doesn't properly set CA lengths which will allow for better rotation of the leaf certificates. 

# TODO

- [x] Add documentation on how to use it and how to check your certificates
- [ ] Add documentation on disaster recovery of all expired certificates(other PR?)
- [x] Think about tests

# How to use

1. Install a cluster and set `certs_validity_period_hours = 1`
2. Test if everything works
3. Set `certs_validity_period_hours = <something bigger>`
4. Run `lokoctl cluster certificate rotate`
5. Check the ca inside the cluster and kubectl for validity
6. Let it run for an hour, if everything still works it means all certs are rotated


# Testing done

- Tested on
  - [x] bare metal VMs
  - [x] AWS
  - [x] Equinix Metal
  - [x] Tinkerbell
